src/aspire/source/image.py

```diff
@@ -345,13 +345,17 @@ def _images(self, start=0, num=np.inf, indices=None):
             "Subclasses should implement this and return an Image object"
         )
 
-    def _apply_filters(self, im_orig, start=0, num=np.inf, indices=None):
+    def _apply_filters(
+        self, im_orig, filters, filter_indices, start=0, num=np.inf, indices=None
+    ):
         """
         For each image in `im_orig` specified by start, num, or indices,
-        the unique_filters associated with the corresponding index in the
-        `ImageSource` are applied. The images are then returned as an `Image`
-        stack.
+        the `filters` associated with the corresponding index in the
```

Comment:
No need to reference `ImageSource`. We're applying the “`filters`  associated with the corresponding index in the supplied `filter_indices`”.